src/Compilers/CSharp/Test/Emit2/CodeGen/CodeGenOptimizedNullableOperators.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGen
+{
+    public class CodeGenOptimizedNullableOperators : CSharpTestBase
+    {
+        [Fact]
+        public void TestComparisonNullableWithNonDefaultConstantValue()
+        {
+            var code = @"
+public class C
+{
+    public bool M(int? x)
+    {
+        return x == 42;
+    }
+}
+";
+            CompileAndVerify(code).VerifyDiagnostics().VerifyIL("C.M", @"
+{
+  // Code size       16 (0x10)
+  .maxstack  2
+  .locals init (int? V_0,
+                int V_1)
+  IL_0000:  ldarg.1
+  IL_0001:  stloc.0
+  IL_0002:  ldc.i4.s   42
+  IL_0004:  stloc.1
+  IL_0005:  ldloca.s   V_0
+  IL_0007:  call       ""int int?.GetValueOrDefault()""
+  IL_000c:  ldloc.1
+  IL_000d:  ceq
+  IL_000f:  ret
+}
+");
+        }
+
+        [Fact]
+        public void TestComparisonNullableWithDefaultConstantValue()
+        {
+            var code = @"
+public class C
+{
+    public bool M(int? x)
+    {
+        return x == 0;
+    }
+}
+";
+            CompileAndVerify(code).VerifyDiagnostics().VerifyIL("C.M", @"
```

Comment:
>CompileAndVerify(code) [](http://example.com/codeflow?start=12&length=22)

Same comment here about verifying runtime behavior. It is probably applicable to other tests in this file as well.